src/check_config_provider/redis_config_provider.rs

```diff
@@ -88,6 +138,94 @@ impl RedisConfigProvider {
             }
         }
     }
+
+    async fn monitor_updates(
+        &self,
+        manager: Arc<Manager>,
+        partitions: &[RedisPartition],
+        shutdown: CancellationToken,
+    ) {
+        let mut conn = self
+            .redis
+            .get_multiplexed_tokio_connection()
+            .await
+            .expect("Unable to connect to Redis");
+        let mut interval = interval(self.check_interval);
+
+        while !shutdown.is_cancelled() {
+            let _ = interval.tick().await;
+
+            for partition in partitions.iter() {
+                let mut pipe = redis::pipe();
+                // We fetch all updates from the list and then delete the key. We do this
+                // atomically so that there isn't any chance of a race
+                let (config_adds, config_deletes): (Vec<_>, Vec<_>) = pipe
+                    .atomic()
+                    .hvals(&partition.update_key)
+                    .del(&partition.update_key)
+                    .query_async::<(Vec<Vec<u8>>, ())>(&mut conn)
+                    .await
+                    .unwrap()
+                    .0 // Get just the LRANGE results
+                    .iter()
+                    .map(|payload| {
+                        rmp_serde::from_slice::<ConfigUpdate>(payload).map_err(|err| {
+                            tracing::error!(?err, "config_consumer.invalid_config_message");
+                            err
+                        })
+                    })
+                    .filter_map(Result::ok)
+                    .partition(|cu| cu.action == ConfigUpdateAction::Add);
+
+                config_deletes.into_iter().for_each(|config_delete| {
+                    manager
+                        .get_service(partition.partition)
+                        .get_config_store()
+                        .write()
+                        .unwrap()
+                        .remove_config(config_delete.subscription_id);
+                    tracing::debug!(
+                        %config_delete.subscription_id,
+                        "config_consumer.config_removed"
+                    );
+                });
```

Comment:
I think you could just use a regular `for` loop here instead of the for each actually

Reply:
Yeah... It's not too clear to me which is prefferable